.github/workflows/build.yml

```diff
@@ -187,6 +187,52 @@ jobs:
         with:
           path: integration-test
 
+
+  trim-analysis:
+    name: Trim analysis
+    runs-on: macos-15
```

Comment:
We pin to get the correct version of Xcode. As of about a month ago, the GitHub macos runners are associated with specific major versions of Xcode. 
- macos-14 has Xcode 15.x installed
- macos-15 has Xcode 16.x installed

I've basically started trying to pin everything (specific runner version, .NET workload versions via the ios18.0 or android35.0 monkiers etc.) since it gets us closer to deterministic builds. If we don't do this, too much changes arbitrarily between different runs against the same commit and CI fails arbitrarily.